tests/qnodes/test_qnode_passthru.py

```diff
@@ -0,0 +1,186 @@
+        theta = tf.Variable(1.7, dtype=self.DTYPE)
+        res = node(phi, theta)
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2,)
+        assert res.dtype == self.DTYPE
+
+    def test_evaluate(self, tensornet_tf_device, tol):
+        """Tests the evaluation of a PassthruQNode."""
+
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            qml.RY(x[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.RX(x[2], wires=1)
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        x = np.array([-1.4, 0.2, 2.1])
+        x_tf = tf.Variable(x, dtype=self.DTYPE)
+
+        node = PassthruQNode(circuit, tensornet_tf_device)
+        res = node(x_tf)
+
+        # compare to a reference node
+        ref_device = qml.device('default.qubit', wires=2)
+        ref_node = BaseQNode(circuit, ref_device)
+        ref_res = ref_node(x)
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2,)
+        assert res.dtype == self.DTYPE
+        assert res.numpy() == pytest.approx(ref_res, abs=tol)
+
+    @pytest.mark.parametrize("vectorize_jacobian", [True, False])
+    def test_jacobian(self, tensornet_tf_device, vectorize_jacobian, tol):
+        """Tests the computing of the Jacobian of a PassthruQNode using TensorFlow."""
+
+        def circuit(phi, theta):
+            qml.RX(phi[0], wires=0)
+            qml.RY(phi[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(theta, wires=0)
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
+
+        ph = np.array([0.7, -1.2])
+        th = 1.7
+        phi = tf.Variable(ph, dtype=self.DTYPE)
+        theta = tf.Variable(th, dtype=self.DTYPE)
+
+        node = PassthruQNode(circuit, tensornet_tf_device)
+        # In TF 2, tf.GradientTape.jacobian comes with a vectorization option.
+        with tf.GradientTape(persistent=not vectorize_jacobian) as tape:
+            tape.watch([phi, theta])
+            res = node(phi, theta)
+        phi_grad, theta_grad = tape.jacobian(res, [phi, theta],
+                                             unconnected_gradients=tf.UnconnectedGradients.ZERO,
+                                             experimental_use_pfor=vectorize_jacobian)
+
+        # compare to a reference Jacobian computed using finite differences
+        ref_device = qml.device('default.qubit', wires=2)
+        ref_node = JacobianQNode(circuit, ref_device)
+        ref_jac = ref_node.jacobian([ph, th])
+
+        assert isinstance(phi_grad, tf.Tensor)
+        assert phi_grad.shape == (2, 2)
+        assert phi_grad.dtype == self.DTYPE
+        assert phi_grad.numpy() == pytest.approx(ref_jac[:, 0:2], abs=tol)
+
+        assert isinstance(theta_grad, tf.Tensor)
+        assert theta_grad.shape == (2,)
+        assert theta_grad.dtype == self.DTYPE
+        assert theta_grad.numpy() == pytest.approx(ref_jac[:, 2], abs=tol)
```

Comment:
This file seemed small at first, so I tried to think of some other basic test cases, but I couldn't come up with any obvious candidates that are not already done. @josh146 any ideas?